hardware/chisel/src/main/scala/core/TensorAlu.scala

```diff
@@ -97,38 +97,330 @@ class AluVector(implicit p: Parameters) extends Module {
+      }
+    }
+
+    // alu
+    val tensorOpBits_piped = ShiftRegister(
+    decSplit0(idx/(numVecUnits/decSplitNb0)).alu_op,
+    if(aluDataReadPipeDelay < 2) aluDataReadPipeDelay else aluDataReadPipeDelay -1)
+    val isSHR = (tensorOpBits_piped === ALU_OP(3))
+    val neg_shift = isSHR & tensorImmBits_piped(C_ALU_IMM_BITS - 1)
+    val fixme_alu_op = Mux(
+      neg_shift,
+      ALU_OP(4), // use opcode = 4 for left shift
+      tensorOpBits_piped)
+    alu.io.opcode := fixme_alu_op
+
+    assert( !valid_003 || io.acc.rd(idx).data.valid)
+
+    alu.io.acc_a.data.valid := RegNext(valid_002) //valid_003 split
+
+    for(aluLenIdx <- 0 until alu.io.acc_a.lenSplit) {
+      for(aluWdtIdx <- 0 until alu.io.acc_a.widthSplit) {
+        val (accGrpIdx, accLenIdx, accWdtIdx) =
+          alu.io.acc_a.reindexDataFromGroup(idx, aluLenIdx, aluWdtIdx)
+        dataRemapA(idx)(aluLenIdx)(aluWdtIdx) :=
+          io.acc.rd(accGrpIdx).data.bits(accLenIdx)(accWdtIdx)
+        alu.io.acc_a.data.bits := dataRemapA(idx)
+      }
+    }
+    val tensorUseImmBits_piped = ShiftRegister(
+    decSplit0(idx/(numVecUnits/decSplitNb0)).alu_use_imm,
+    if(aluDataReadPipeDelay < 2) aluDataReadPipeDelay else aluDataReadPipeDelay -1)
+    alu.io.acc_b.data.valid := Mux(tensorUseImmBits_piped,
+      tensorImm.data.valid,
+      valid_003)
+    alu.io.acc_b.data.bits := Mux(tensorUseImmBits_piped,
+      tensorImm.data.bits,
+      save_src)
+
+    assert( alu.io.acc_y.data.valid === valid_004)
+    io.acc.wr(idx).valid := RegNext(valid_003) //valid_004 split
+    io.acc.wr(idx).bits.idx := RegNext(dst_idx_003)//dst_idx_004 split
+
+    for(aluLenIdx <- 0 until alu.io.acc_y.lenSplit) {
+      for(aluWdtIdx <- 0 until alu.io.acc_y.widthSplit) {
+        val (accGrpIdx, accLenIdx, accWdtIdx) =
+          alu.io.acc_y.reindexDataFromGroup(idx, aluLenIdx, aluWdtIdx)
+        io.acc.wr(accGrpIdx).bits.data(accLenIdx)(accWdtIdx) :=
+          alu.io.acc_y.data.bits(aluLenIdx)(aluWdtIdx)
+      }
+    }
+
+    assert( alu.io.out.data.valid === valid_004)
+    for (idx1 <- 0 until io.out.tensorLength) {
+      for (idx2 <- 0 until io.out.tensorWidth/numVecUnits) {
+        outData(idx1)(idx*io.out.tensorWidth/numVecUnits + idx2) := alu.io.out.data.bits(idx1)(idx2)
+      }
+    }
+  }
+
+// comment for split write
+  io.out.wr(0).valid := valid_004
+  io.out.wr(0).bits.idx := dst_idx_004
+  io.out.wr(0).bits.data := outData
+  io.out.tieoffRead()
+
+  val bypass_dst = valid_003 && valid_004 && ( dst_idx_004 === dst_idx_003)
+  val bypass_src = src_valid_003 && valid_004 && ( dst_idx_004 === src_idx_003)
+
+  // Do we need a bypass
+  when ( bypass_dst) {
+    printf( "Bypass required on dst_idx read %x RAW with write %x\n", dst_idx_003, dst_idx_004)
+    assert( false.B, "DST bypass required")
+  }
+  when ( bypass_src) {
+    printf( "Bypass required on src_idx read %x RAW with write %x\n", src_idx_003, dst_idx_004)
+    assert( false.B, "SRC bypass required")
+  }
+}
```

Comment:
why not doing? `assert(!bypass_src, ...)`

Reply:
extracted asserts as suggested